src/main/java/core/basesyntax/model/zoo/Animal.java

```diff
@@ -1,6 +1,21 @@
 package core.basesyntax.model.zoo;
 
+import javax.persistence.DiscriminatorColumn;
+import javax.persistence.DiscriminatorType;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Inheritance;
+import javax.persistence.InheritanceType;
+
+@Entity
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+@DiscriminatorColumn(name = "animal_type", discriminatorType = DiscriminatorType.STRING)
 public class Animal {
+    @Id
+    @GeneratedValue(strategy = GenerationType.TABLE)
+    private Long id;
```

Comment:
```suggestion
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
```
add getter and setter